src/components/ui/DrawerItem.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useState } from "react";
+import { createRoot } from "react-dom/client";
+import {
+  ListItemButton,
+  ListItemIcon,
+  Icon,
+  ListItemText,
+} from "@mui/material";
+import DragIndicatorIcon from "@mui/icons-material/DragIndicator";
+import { combine } from "@atlaskit/pragmatic-drag-and-drop/combine";
+import {
+  draggable,
+  dropTargetForElements,
+} from "@atlaskit/pragmatic-drag-and-drop/element/adapter";
+import { DropIndicator } from "@atlaskit/pragmatic-drag-and-drop-react-drop-indicator/box";
+import {
+  Edge,
+  attachClosestEdge,
+  extractClosestEdge,
+} from "@atlaskit/pragmatic-drag-and-drop-hitbox/closest-edge";
+import { setCustomNativeDragPreview } from "@atlaskit/pragmatic-drag-and-drop/element/set-custom-native-drag-preview";
+import { ThemeProvider as MUIThemeProvider } from "@mui/material/styles";
+import { ThemeProvider } from "@/contexts/ThemeProvider";
+import { useMUITheme } from "@/hooks/theme";
+import { useSearchParamsUtils } from "@/hooks/searchParamsUtils";
+
+/**
+ * Represents a drawer item component.
+ *
+ * @param item - The item object containing the title, id, and icon.
+ * @param index - The index of the item.
+ * @param groupTitle - The title of the group.
+ * @returns The rendered JSX for the drawer item.
+ */
+export default function DrawerItem({
+  item: { title, id, icon },
+  index,
+  groupTitle,
+}: {
+  item: { title: string; id: string; icon: React.ComponentType };
+  index: number;
+  groupTitle: string;
+}) {
+  // Ref to use for the draggable element
+  const dragRef = React.useRef(null);
+  // Ref to use for the handle of the draggable element, must be a child of the draggable element
+  const handleRef = React.useRef(null);
+  const theme = useMUITheme();
+  const { setParam } = useSearchParamsUtils();
+  // Represents the closest edge to the mouse cursor
+  const [closestEdge, setClosestEdge]: any = useState<Edge | null>(null);
+
+  useEffect(() => {
+    if (!dragRef.current || !handleRef.current) return;
+    const element = dragRef.current;
+    const handleItem = handleRef.current;
+
+    return combine(
+      // makes the item draggable
+      draggable({
+        element: element,
+        dragHandle: handleItem,
+        // Sets the initial data for the drag and drop interaction
+        getInitialData: () => ({ index, title: groupTitle }),
+        // Sets a lightweight version of the real item as a preview
+        onGenerateDragPreview: ({ nativeSetDragImage, source, location }) => {
+          setCustomNativeDragPreview({
+            nativeSetDragImage,
+            render: ({ container }) => {
+              const root = createRoot(container);
+              root.render(
+                <ThemeProvider>
+                  <MUIThemeProvider theme={theme}>
```

Comment:
As the preview is created in a new root, it does not have access to all the contexts provided by React.
In order to have styles applied correctly to the preview, it must be placed inside `ThemeProvider` and `MUIThemeProvider`.

As an alternative, it is possible to use React's createPortal to copy all contexts but it might be costly.